primitives/src/lib.rs

```diff
@@ -28,19 +28,37 @@ extern crate std;
 extern crate actual_serde as serde;
 
 #[cfg(feature = "alloc")]
-pub mod locktime;
+pub mod locktime; // Maintains units module structure and re-exports everything.
 pub mod opcodes;
 pub mod sequence;
 
-#[rustfmt::skip]                // Keep public re-exports separate.
 #[doc(inline)]
 #[cfg(feature = "alloc")]
-pub use self::{
-    locktime::{absolute, relative},
-};
+pub use self::locktime::{absolute, relative};
 #[doc(inline)]
 pub use self::sequence::Sequence;
 
+// Re-export everything from the `units` crate.
+//
+// Users of `primitives` should not need go to `units`. N.B.
+// - This is important for correct access to `unit` types, it has to be correct.
+// - Modules are needed for niche error usage and also exposed for ergonomics (eg, `module::Error`)
+#[rustfmt::skip]                // Keep public unit re-exports separate.
+#[doc(inline)]
+pub use units::amount::{self, Amount, SignedAmount};
+#[cfg(feature = "alloc")]
+#[doc(inline)]
+pub use units::{
+    // Re-export everything.
+    block::{self, BlockHeight, BlockInterval, TooBigForRelativeBlockHeightError},
+    // Re-export everything.
+    fee_rate::FeeRate,
+    // Do not re-export: MissingPrefixError, ContainsPrefixError
+    parse::{self, ParseIntError, PrefixedHexError, UnprefixedHexError},
```

Comment:
Agreed. Let's stick an `error` submodule and dump all the error types into there. (And maybe we should do this consistently across crates when we do mass-re-exports like this.)

Reply:
Seems re-export all these in `bitcoin` already since https://github.com/rust-bitcoin/rust-bitcoin/issues/2874 was resolved in https://github.com/rust-bitcoin/rust-bitcoin/pull/2895

> Let's stick an error submodule and dump all the error types into there.

A submodule of what module? I we don't want them at the crate root we can just re-export `parse` and folks can got to `bitcoin::parse::PrefixedHexError`. 

FTR when thinking about the `pares` module I found it slightly different to say `block::FooError` because `parse` is kind of less descriptive.

Reply:
I created an issue for exactly this thread of discussion: https://github.com/rust-bitcoin/rust-bitcoin/issues/3068

Reply:
Changing to use wildcard after consensus in #3068.